From e9cee8da8eeb60565eafa2d1f83b144f02fff195 Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Fri, 28 Oct 2011 19:15:41 +0800
Subject: [PATCH 043/641] UniCore64: Add some functions about power

Fixed error:
elfcore.c:(.text+0x231bc): undefined reference to `machine_power_off'
elfcore.c:(.text+0x233c8): undefined reference to `machine_halt'
elfcore.c:(.text+0x23430): undefined reference to `machine_restart'
elfcore.c:(.text+0x23740): undefined reference to `pm_power_off'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/Makefile  |    2 +-
 arch/unicore64/kernel/process.c |   34 ++++++++++++++++++++++++++++++++++
 2 files changed, 35 insertions(+), 1 deletion(-)
 create mode 100644 arch/unicore64/kernel/process.c

diff --git a/arch/unicore64/kernel/Makefile b/arch/unicore64/kernel/Makefile
index 468fc2b..5e50d9c 100644
--- a/arch/unicore64/kernel/Makefile
+++ b/arch/unicore64/kernel/Makefile
@@ -1,3 +1,3 @@
-obj-y			:= tlb.o ptrace.o setup.o init_task.o
+obj-y			:= tlb.o ptrace.o setup.o init_task.o process.o
 head-y			:= head.o
 extra-y			:= $(head-y) vmlinux.lds
diff --git a/arch/unicore64/kernel/process.c b/arch/unicore64/kernel/process.c
new file mode 100644
index 0000000..1e44fd1
--- /dev/null
+++ b/arch/unicore64/kernel/process.c
@@ -0,0 +1,34 @@
+#include <linux/kernel.h>
+
+/**
+ * machine_restart() - Reboot the machine
+ * @cmd:
+ */
+void machine_restart(char *cmd)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * machine_halt() - Halt the machine
+ */
+void machine_halt(void)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * machine_power_off() - Power off the machine
+ */
+void machine_power_off(void)
+{
+	/* FIXME */
+	BUG();
+}
+
+/**
+ * Function pointers to optional machine specific functions
+ */
+void (*pm_power_off)(void) = NULL;
-- 
1.7.9.5

